modes/play/levels/level.js

```diff
@@ -21,7 +21,32 @@ class Level {
     return { x, y };
   }
 
-  generateMobs(config) {
-    return config.mobs && config.mobs.map(mob => new mob.type(this.game, this.mode, ...mob.coords))
+  processGrid(config) {
+    var playerSpawn = [50, 50];  // hilariously dumb default
+    var mobs = [];
+    var ground = [];
+    var scale = config.scale || [10, 10];
+    var grid = config.grid.trim().split('\n');
+    for (var row in grid) {
+      var gridLine = grid[row];
+      for (var col in gridLine) {
+        var cell = gridLine[col];
+        if (cell === '.') continue;
+        var x = +col * scale[0];
+        var y = +row * scale[1];
+        // NOTA BENE: The following conditions are non-exclusive.  I believe that to be a feature.
```

Comment:
Yeah, the way this happened is that it was making me uncomfortable that my level format didn''t prohibit overlap.  Obviously I could prohibit overlap in the code you've highlighted here, but that felt like it would an opportunity for weird bugs in the level format.  So I was contemplating it, and then I realized that hey, maybe it'd be useful to be like this anyway.

One example I thought of was this: one day, maps should probably have `regions`, which are not mobs or blocks, but are areas that apply some effect to mobs within them.  Like "water", but also like "updraft" or "speed booster" or whatever.  Those should obviously be possible to overlap with mobs and blocks.